### PR TITLE
feat: Introduce Device AHS validator for additional device information.

### PR DESCRIPTION
**Pull request by Altanali**
*Issue #, if available:*

*Description of changes:*
Extend the AHS Validator by introducing a DeviceCapabilitiesConstants data class and additional Device-AtomArrangement, LocalDetuning, Hamiltonian, and DrivingField validators for information that's available in AWS AHS devices (e.g. Quera Aquila). 

*Testing done:*

Testing done to achieve full coverage. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: a.txt
Comment (math411) [inline a.txt:1]: This can be removed

File: src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from braket.analog_hamiltonian_simulator.rydberg.validators.device_validators.device_atom_arrangement import (  # noqa: E501 F401
+    DeviceAtomArrangementValidator,
+)
+from braket.analog_hamiltonian_simulator.rydberg.validators.device_validators.device_capabilities_constants import (  # noqa: E501 F401
+    DeviceCapabilitiesConstants,
```
Comment (math411) [inline src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/__init__.py:5]: These two imports are going to cause circular import issues. One fix would be to swap the ordering of L1 and L4 but I would encourage you to see what other options there would be to resolve the circular import issue.

File: src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_atom_arrangement.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    # The number of filled lattice sites must not exceed MAX_FILLED_SITES.
+    @root_validator(pre=True, skip_on_failure=True)
+    def atom_number_limit(cls, values):
+        filling = values["filling"]
+        capabilities = values["capabilities"]
+        qubits = sum(filling)
+        if qubits > capabilities.MAX_FILLED_SITES:
+            raise ValueError(
+                f"Filling has {qubits} '1' entries; is must have not\
```
Comment (math411) [inline src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_atom_arrangement.py:87]:
```suggestion
                f"Filling has {qubits} '1' entries; it must have not\
```

File: src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_driving_field.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    def amplitude_start_and_end_values(cls, values):
+        amplitude = values["amplitude"]
+        time_series = amplitude["time_series"]
+        time_series_values = time_series["values"]
+        if time_series_values:
+            start_value, end_value = time_series_values[0], time_series_values[-1]
+            if start_value != 0 or end_value != 0:
+                raise ValueError(
+                    f"The values of the Rabi frequency at the first and last time points are \
+                        {start_value}, {end_value}; they both must be both 0."
```
Comment (math411) [inline src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_driving_field.py:33]:
```suggestion
                        {start_value}, {end_value}; they both must be nonzero."
```

File: src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_hamiltonian.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+class DeviceHamiltonianValidator(HamiltonianValidator):
+    LOCAL_RYDBERG_CAPABILITIES: bool = False
+
+    @root_validator(pre=True, skip_on_failure=True)
+    def max_zero_local_detuning(cls, values):
+        LOCAL_RYDBERG_CAPABILITIES = values["LOCAL_RYDBERG_CAPABILITIES"]
+        local_detuning = values.get("localDetuning", [])
+        if not LOCAL_RYDBERG_CAPABILITIES:
+            if len(local_detuning) > 0:
```
Comment (math411) [inline src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/device_hamiltonian.py:14]:
```suggestion
        if not LOCAL_RYDBERG_CAPABILITIES and len(local_detuning):
```

File: a.txt
Comment (Altanali) [reply a.txt:1]: Removed!

File: src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/__init__.py (hunk 1 above)
Comment (Altanali) [reply src/braket/analog_hamiltonian_simulator/rydberg/validators/device_validators/__init__.py]: In 746ab2c I opt to move `device_capabilities_constants.py` to the parent directory, so it lives alongside the `capabilities_constants.py`; this does break the separation of `device_validation` and `validation` constructs/methods but separates the `DeviceCapabilitiesConstants` from the constructs that use it without having to enforce an import order in `__init__.py`. 

Please let me know if this approach is appropriate!

**Comment by Altanali** [comment]
Closing - Moving AHS Device Validation to the BDK instead of extending the default-simulator repo.

**Pull request closed without merging**